workflow/common/util.go

```diff
@@ -257,8 +257,8 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.Arguments, globalParams, localPa
 			for _, argParam := range argParams {
 				newName := argParam.Name
 				newValue := argParam.Value
-				newPassthroughRegexp := argParam.PassthroughRegexp //used for workflow templates where all arguments may not have been resolved already
-				newParameter := wfv1.Parameter{Name: newName, Value: newValue, PassthroughRegexp: newPassthroughRegexp}
+				newRegexp := argParam.PassthroughRegexp //used for workflow templates where all arguments may not have been resolved already
```

Comment:
I don't fully understand why `Regexp` is copied from `PassthroughRegexp`. Could you tell me an example manifests which describe the necessity of this copy?

Reply:
TestPassthroughParams in operator_test.go fails without it - where we have arguments using a passthroughRegexp being passed into a step with regex inputs.

Reply:
https://github.com/argoproj/argo/pull/1490#issuecomment-515494144
Please try overwriting `tmpl`.

Reply:
Hmm, changing it causes another error. I'm trying to fix all of them.

Reply:
I fixed the validation code in https://github.com/argoproj/argo/pull/1508.

However, I realized we should use `scope` of called templates instead of overwriting arguments of caller templates.